pkg/validation/validate_cluster.go

```diff
@@ -195,22 +198,32 @@ func (v *ValidationCluster) collectComponentFailures(client kubernetes.Interface
 }
 
 func (v *ValidationCluster) collectPodFailures(client kubernetes.Interface) error {
-	pods, err := client.CoreV1().Pods("kube-system").List(metav1.ListOptions{})
-	if err != nil {
-		return fmt.Errorf("error listing Pods: %v", err)
-	}
-
-	for _, pod := range pods.Items {
-		if pod.Status.Phase == v1.PodSucceeded {
-			continue
+	err := pager.New(pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
+		return client.CoreV1().Pods(metav1.NamespaceAll).List(opts)
+	})).EachListItem(context.TODO(), metav1.ListOptions{}, func(obj runtime.Object) error {
+		pod := obj.(*v1.Pod)
+		priority := pod.Spec.PriorityClassName
+		if priority != "system-cluster-critical" && priority != "system-node-critical" {
+			return nil
+		}
+		if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
```

Comment:
Why is `PodFailed` skipped? Shouldn't that be a validation error?

Reply:
It is unlikely to change status just by waiting. Like `PodSucceeded`, the pod is spent.

It also wasn't a validation error before.

Reply:
I created a failed pod and looking at its [status](https://gist.github.com/rifelpet/ea3938ddbfe181ab327ed08e7fd7823a) and the [original code](https://github.com/kubernetes/kops/blob/cbee4dccbaa1a6c0490a671500db921c34b521b7/pkg/validation/validate_cluster.go#L216-L228) I would expect the container's status of NotReady to cause a validation error.

Reply:
Okay, I walked that part back